### PR TITLE
Split Coverage Test into Multiple Test Cases

### PR DESCRIPTION
**Pull request by SlicedSilver**
**Type of PR:** enhancement (e2e tests)

**PR checklist:**
- [ ] Addresses an existing issue
- [x] Includes tests
- [ ] Documentation update

**Overview of change:**

Instead of a single test script for the e2e coverage test which had grow large and complicated, the coverage test is now split into smaller test cases. This should make it easier to cover future API changes in the future.

The coverage score has been increased by 1% to 92%.

**Note:**
This builds upon the #1167 PR and includes those changes.

### CODE REVIEW

File: src/gui/chart-widget.ts [hunk 1]
```diff
@@ -406,6 +415,18 @@ export class ChartWidget implements IDestroyable {
 		}
 	}
 
+	private _shouldSubscribeMouseWheel(options: DeepPartial<ChartOptionsInternal>): boolean {
+		return Boolean(options?.handleScroll?.mouseWheel || options?.handleScale?.mouseWheel);
+	}
```
Comment (kirchet) [inline src/gui/chart-widget.ts:420]: I think `_shouldSubscribeMouseWheel` should be a function instead of a method

File: src/gui/chart-widget.ts [hunk 2]
```diff
@@ -406,6 +415,18 @@ export class ChartWidget implements IDestroyable {
 		}
 	}
 
+	private _shouldSubscribeMouseWheel(options: DeepPartial<ChartOptionsInternal>): boolean {
+		return Boolean(options?.handleScroll?.mouseWheel || options?.handleScale?.mouseWheel);
```
Comment (kirchet) [inline src/gui/chart-widget.ts:419]: We can change the type of options to `ChartOptionsInternal` and get rid of `.?` operators

```ts
const currentlyHasMouseWheelListener = this._shouldSubscribeMouseWheel(this._options);
this._model.applyOptions(options);
const shouldHaveMouseWheelListener = this._shouldSubscribeMouseWheel(this._options);
```

File: src/gui/chart-widget.ts (hunk 1 above)
Comment (SlicedSilver) [reply src/gui/chart-widget.ts:420]: Will make the change 👍

File: src/gui/chart-widget.ts (hunk 2 above)
Comment (SlicedSilver) [reply src/gui/chart-widget.ts:419]: @kirchet resolved on the #1167 PR.

Comment (kirchet) [reply src/gui/chart-widget.ts:419]: OK, then we need to merge the https://github.com/tradingview/lightweight-charts/pull/1167 PR first and then merge `master` into this branh

**Review by kirchet — APPROVED** [review]

**Review by romfrancois — APPROVED** [review]

**Pull request merged**